src/main/java/net/reini/rabbitmq/cdi/ConnectionManagerWatcherThread.java

```diff
@@ -0,0 +1,71 @@
+package net.reini.rabbitmq.cdi;
+
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConnectionManagerWatcherThread extends Thread {
```

Comment:
I would prefer a an solution using an `ExecutorService` / `Runnable` here, because spinning up Thread in a JavaEE container should be avoided but there we could later use an container managed executor...

Reply:
I had a look at this before adding this as long as the class is not managed like ejb classes it should cause no problem. Rabbit library is also spawning own threads. For managed classes even sleep wait notify synchronized and all sorts of locks are no good style as this also counts as managing threads. Which is not recommended inside jee container managed classes. Of course i could change this is you insist on it

Reply:
Additional explanation to my initial decision:
If we use an ExecutorService provided by the container we will run in a managed thread which should not be managed by ourself. (no locks, notify,wait)
i think the alternative to manage an unmanged thread is better.
For that reason i decided to use a plain old thread instead of ExecutorService.